File: lib-database/src/main/java/com/ustadmobile/core/db/dao/AccessTokenDao.java
package com.ustadmobile.core.db.dao;

import com.ustadmobile.lib.database.annotation.UmDao;
import com.ustadmobile.lib.database.annotation.UmQuery;

@UmDao
public abstract class AccessTokenDao {

    @UmQuery("SELECT EXISTS(SELECT token FROM AccessToken WHERE accessTokenPersonUid = :personUid AND token = :token)")
    public abstract boolean isValidToken(long personUid, String token);

}

File: lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/AccessTokenDao.kt
package com.ustadmobile.core.db.dao

import androidx.room.Dao
import androidx.room.Query

@Dao
abstract class AccessTokenDao {

    @Query("SELECT EXISTS(SELECT token FROM AccessToken WHERE accessTokenPersonUid = :personUid AND token = :token)")
    abstract fun isValidToken(personUid: Long, token: String): Boolean

}
